src/components/Toaster/__tests__/ToasterProvider.test.tsx

```diff
@@ -302,3 +303,113 @@ describe('api.has', () => {
         expect(providerAPI.has(toastProps.name)).toBe(false);
     });
 });
+
+describe('interaction with Modal while closing Toaster', () => {
+    const MODAL_CONTENT = 'qwerty-modal';
+
+    const ModalAPI = ({onMount}: {onMount: (openModalFun: () => void) => any}) => {
+        const [open, setOpenFlag] = React.useState(true);
+        const closeModal = () => setOpenFlag(false);
+
+        React.useEffect(() => {
+            onMount(() => setOpenFlag(true));
+        }, []);
+
+        return (
+            <Modal open={open} onOutsideClick={closeModal}>
+                {MODAL_CONTENT}
+            </Modal>
+        );
+    };
+
+    function setup() {
```

Comment:
I think we can simplify setup to pass what should be opened initially -- modal or toaster, and add button to modal, that will open toaster via hook

Reply:
such property more likely complicate the code cause I will have to write property logic inside `setup` without being able to check that the Modal has opened.

like

now I can write this
```
await setup()

await openModal()
await expectModalOpened()

await openToast()
await expectToastOpened()
```
and I am absolutely sure that Modal was opened and opened before Toaster was

while in property case it will be

```
await setup(FirstOpenModal) 
// How I can really be sure that Modal was opened before Toaster 
// if i can not call expectModalOpened and expectToastOpened inside setup 
// and I am testing Modal and Toaster (I do not trust Modal and Toaster)
// I can only check that they are opened

await expectModalOpened()
await expectToastOpened()
```

and moreover, the code will only get bigger and more difficult

```
// now
await openModal()
await openToast()
...

or

await openToast()
await openModal()
...
```

it will be in property case

```
if (FirstOpenModal) {
    openModal()
    openToast()
} else {
    openToast()
    openModal()
}
```